### PR TITLE
Bug protobuf fallback logic

### PR DESCRIPTION
**Pull request by michax**
Fix for protobuf logic.

## Problem
Currently when endpoint doesn't support protobuf, endpoint returns an error and openapi-clientlib fallbacks to json subscription.
However in addition to a subscribe, we also send an error message which in our GO sagas drops/locks subscription from further passing updates.

## Solution
To be backward compatible, I've dropped invoking error callbacks when endpoint fails due to protobuf format being unsupported.
This ensure that userspace can keep current error handling logic with duplicated code need for handling protobuf errors.
So in short. Due to error being inreversible/fatal one, we only now send it whenever subscription fails without fallback. However in fallback scenarios fatal error isn't sent.

## Alternative options

If we decide to keep error being invoked, we should then drop re-connect logic as well and require it to be implemented on user space side.

### CODE REVIEW

File: src/openapi/streaming/subscription.spec.js [hunk 1]
```diff
@@ -154,8 +154,6 @@ describe('openapi StreamingSubscription', () => {
             transport.post.mockClear();
 
             setTimeout(() => {
-                expect(errorSpy.mock.calls.length).toEqual(1);
```
Comment (lukeapage) [inline src/openapi/streaming/subscription.spec.js]: we should test its *not* called?

Comment (michax) [reply src/openapi/streaming/subscription.spec.js]: @lukeapage Good point. Done.
